OpenRA.Mods.Common/Util.cs

```diff
@@ -155,5 +158,70 @@ public static IEnumerable<CPos> RandomWalk(CPos p, MersenneTwister r)
 				yield return p;
 			}
 		}
+
+		public static string FriendlyTypeName(Type t)
+		{
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(HashSet<>))
+				return "Set of {0}".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
+
+			if (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return "Dictionary<{0},{1}>".F(t.GetGenericArguments().Select(FriendlyTypeName).ToArray());
```

Comment:
The Dictionary and Cached types.  Right now you have "2D Integer"s and "List of foo"s, but then "Dictionary<foo, bar>" which is completely inconsistent.